modules/users/client/controllers/admin/add-user.client.controller.js

```diff
@@ -27,25 +21,30 @@
         return false;
       }
 
-      UsersService.userSignup(vm.credentials)
+      vm.credentials.roles = alterRole(vm.credentials.roles);
+
+      ApplicantsService.adminSignup(vm.credentials)
         .then(onAddUserSuccess)
         .catch(onAddUserError);
     }
 
-    // Authentication Callbacks
-
     function onAddUserSuccess(response) {
-      // If successful we assign the response to the global user model
-      //vm.authentication.user = response;
       Notification.success({ message: '<i class="glyphicon glyphicon-ok"></i> Signup successful!' });
-      // And redirect to the previous or home page
       $state.go($state.previous.state.name || 'home', $state.previous.params);
     }
 
     function onAddUserError(response) {
       Notification.error({ message: response.data.message, title: '<i class="glyphicon glyphicon-remove"></i> Add User Error!', delay: 6000 });
     }
 
-    
+    function alterRole (role) {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase 

use this function

Reply:
Changing to lowercase or uppercase is not the only problem. The roles shown on the front end are different from the roles in the schema.

Reply:
Hey, in an HTML dropdown you can have a different value than the text that is shown, like the following:
<option value="ta">TA</option>

Hope this helps, we really shouldn't need to parse it in a seperate JS function. If you're still having trouble doing it the easy way, take a look at my new reassignFromApproval branch for how I did it.